subsys/dfu/src/dfu_target_modem.c

```diff
@@ -65,13 +65,12 @@ static int delete_banked_modem_fw(void)
 
 	while (true) {
 		err = getsockopt(fd, SOL_DFU, SO_DFU_OFFSET, &offset, &len);
-		if (err != 0) {
-			if (err == ENOEXEC) {
+		if (err < 0) {
+			if (errno == ENOEXEC) {
```

Comment:
Does this compile? `errno` here and `err` declared?

Reply:
`errno` is unfortunately a global 🗡👎 which gets set for each call to the modem.

Reply:
so yes it compiles. We can't change the design decision of bsdlib/modem